src/crypto/finger.rs

```diff
@@ -0,0 +1,51 @@
+/// Certificate fingerprint.
+///
+/// DTLS uses self signed certificates, and the fingerprint is communicated via
+/// SDP to let the remote peer verify who is connecting.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct Fingerprint {
+    /// Hash function used to produce the `bytes`.
+    ///
+    /// This is normally `sha-256`.
+    pub hash_func: String,
+
+    /// Digest of the certificate by the algorithm in `hash_func`.
+    pub bytes: Vec<u8>,
+}
+
+impl ToString for Fingerprint {
```

Comment:
I think it is recommended to implement `fmt::Display` instead and rely on the blanket impl to get `ToString`.

Reply:
See https://doc.rust-lang.org/std/string/trait.ToString.html.

Reply:
The format here is very specific (to fit in an SDP). The Display trait won't be right either, we should change this to a function.